core/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java

```diff
@@ -53,4 +118,97 @@ public static void main(String[] args) throws StartupError {
     static void close(String[] args) throws IOException {
         Bootstrap.stop();
     }
+
+    @Argument(handler = SubCommandHandler.class, metaVar = "command")
+    @SubCommands({
+        @SubCommand(name = "start", impl = StartCommand.class),
+        @SubCommand(name = "version", impl = VersionCommand.class)
+    })
+    Command command;
+
+    static abstract class Command {
+
+        @Option(name = "-h", aliases = "--help", usage = "print this message")
```

Comment:
I think `ant` is a bad example (all of the Java tools are non-unix-y), but I'm totally +0 on adding it, if it's not too much trouble. I think it helps the "principal of least surprise":

```
~/es/elasticsearch:master λ ls --version
ls (GNU coreutils) 8.24
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Richard M. Stallman and David MacKenzie.

~/es/elasticsearch:pr/16791 λ gdb --version
GNU gdb (GDB) Fedora 7.10.1-30.fc23
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-redhat-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word".

~/es/elasticsearch:pr/16791 λ cat --version
cat (GNU coreutils) 8.24
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Torbjörn Granlund and Richard M. Stallman.

~/es/elasticsearch:pr/16791 λ emacs --versio
GNU Emacs 25.0.90.1
Copyright (C) 2016 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.

~/es/elasticsearch:pr/16791 λ less --version
less 481 (POSIX regular expressions)
Copyright (C) 1984-2015  Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Homepage: http://www.greenwoodsoftware.com/less

~/es/elasticsearch:pr/16791 λ gcc --version
gcc (GCC) 5.3.1 20151207 (Red Hat 5.3.1-2)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

~/es/elasticsearch:pr/16791 λ systemctl --version
systemd 222
+PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ -LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN
```